lib/src/cache/cache.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'package:w_common/disposable.dart';
+import 'package:w_common/func.dart';
+
+/// Immutable payload that indicates a change in a [Cache].
+class CacheContext<TIdentifier, TValue> {
```

Comment:
Hold off, I might have misunderstood how this was used (because I hadn't read the rest of the code yet...)